bitcoin/examples/bip32.rs

```diff
@@ -48,8 +48,11 @@ fn main() {
 
     // generate first receiving address at m/0/0
     // manually creating indexes this time
-    let zero = ChildNumber::ZERO_NORMAL;
-    let public_key = xpub.derive_pub(&secp, &[zero, zero]).unwrap().public_key;
```

Comment:
Yes, but we already have it. 

`NormalChildNumber` is meant to only be used inside the `NormalChildIterator` to avoid confusion while using bip32 at all.

Reply:
Then the code really should just change the type of `zero`.

> `NormalChildNumber` is meant to only be used inside the `NormalChildIterator`

Nope, it should be used wherever people need to guarantee unhardened paths.

Reply:
Well, if thats the idea.